kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -345,6 +353,19 @@ func (c *ipamController) handleNodeUpdate(kvp model.KVPair) {
 	}
 }
 
+// handleClusterInformationUpdate wraps the logic to execute when receiving a clusterinformation update.
+func (c *ipamController) handleClusterInformationUpdate(kvp model.KVPair) {
+	if kvp.Value != nil {
+		ci := kvp.Value.(v3.ClusterInformation)
+		if ci.Spec.DatastoreReady != nil {
+			c.datastoreReady = *ci.Spec.DatastoreReady
+		}
+	} else {
+		// if there's no ClusterInformation CR object, should we still make the GC run ?
+		c.datastoreReady = false
```

Comment:
Not really sure which one to set here. What that means if we received none clusterinformation CR object?

Reply:
I think this is correct - if there is no cluster info we should assume the datastore is not ready until we're told otherwise. 

There should always be a clusterinfo resource created.